scene/gui/graph_edit.cpp

```diff
@@ -1646,6 +1646,585 @@ HBoxContainer *GraphEdit::get_zoom_hbox() {
+	float gap_v = 100.0f * zoom;
+	float gap_h = 300.0f * zoom;
+
+	//Creating a map with each graph node pointing to a set of its upper neighbours
+	for (int i = get_child_count() - 1; i >= 0; i--) {
+		GraphNode *gn = Object::cast_to<GraphNode>(get_child(i));
+		if (!gn) {
+			continue;
+		}
+
+		if (gn->is_selected()) {
+			selected_nodes.insert(gn->get_name());
+			origin = origin < gn->get_position_offset() ? origin : gn->get_position_offset();
+			Set<StringName> s;
+			for (List<Connection>::Element *E = connections.front(); E; E = E->next()) {
+				GraphNode *p_from = Object::cast_to<GraphNode>(node_names[E->get().from]);
+				if (E->get().to == gn->get_name() && p_from->is_selected()) {
+					if (!s.has(p_from->get_name())) {
+						s.insert(p_from->get_name());
+					}
+					String s_connection = String(p_from->get_name()) + " " + String(E->get().to);
+					StringName _connection(s_connection);
+					Pair<int, int> ports(E->get().from_port, E->get().to_port);
+					if (port_info.has(_connection)) {
+						Pair<int, int> p_ports = port_info[_connection];
+						if (p_ports.first < ports.first) {
+							ports = p_ports;
+						}
+					}
+					port_info.set(_connection, ports);
+				}
+			}
+			upper_neighbours.set(gn->get_name(), s);
+		}
+	}
+
+	HashMap<int, Vector<StringName>> l = layering(selected_nodes, upper_neighbours);
+	crossing_minimisation(l, upper_neighbours);
+
+	Dictionary root, align, sink, shift;
+	horizontal_alignment(root, align, l, upper_neighbours, selected_nodes);
+
+	HashMap<StringName, Vector2> new_positions;
+	Vector2 default_position(FLT_MAX, FLT_MAX);
+	Dictionary inner_shift;
+	Set<StringName> block_heads;
+
+	for (const Set<StringName>::Element *E = selected_nodes.front(); E; E = E->next()) {
+		inner_shift[E->get()] = 0;
+		sink[E->get()] = E->get();
+		shift[E->get()] = FLT_MAX;
+		new_positions.set(E->get(), default_position);
+		if ((StringName)root[E->get()] == E->get()) {
+			block_heads.insert(E->get());
+		}
+	}
+
+	calculate_inner_shifts(inner_shift, root, node_names, align, block_heads, port_info);
+
+	List<Pair<StringName, Connection>> pending_blocks;
+	//call place_block() on each block head
+	for (const Set<StringName>::Element *E = selected_nodes.front(); E; E = E->next()) {
+		place_block(E->get(), gap_v, l, root, align, node_names, inner_shift, sink, shift, new_positions, pending_blocks);
+	}
+
+	//call post_process()
+
+	//Setting new positions of block heads
+	int block_number = 0;
+	for (const Set<StringName>::Element *E = block_heads.front(); E; E = E->next()) {
+		Vector2 cal_pos = new_positions[E->get()];
+		//Finding the layer it block head belongs to
+		unsigned int layer = 0;
+		for (; layer < l.size(); layer++) {
+			Vector<StringName> layer_nodes = l[layer];
+			if (layer_nodes.has(E->get()))
+				break;
+		}
+		cal_pos.x = origin.x + (real_t)(layer)*gap_h;
+		new_positions.set(E->get(), cal_pos);
+		block_number++;
+	}
+	//Now finding positions of nodes inside a block
```

Comment:
Coding style.

> //Now finding positions of nodes inside a block

> // Now finding positions of nodes inside a block.

Reply:
Oops! My bad. I'll add comments as per the guidelines